File: ocaml-swipl/c_tests/atoms.c
#include <stdio.h>
#include <SWI-Prolog.h>

int main(int argc, char **argv) {
  printf("initialise -> %i\n", PL_initialise(argc, argv));

  atom_t atom1 = PL_new_atom("my_atom");
  atom_t atom2 = PL_new_atom("my_atom");
  term_t node1 = PL_new_term_ref();
  term_t node2 = PL_new_term_ref();
  PL_put_atom(node1, atom1);
  PL_put_atom(node2, atom2);

  printf("unify -> %i\n", PL_unify(node1, node2));

  return 0;
}

File: ocaml-swipl/tests/atoms.ml
open Swipl

let atom = "my_atom"

let _ =
  Embedding.initialise [|Sys.argv.(0); "-q"|];
  let atom1 = Atom.make atom in
  let atom2 = Atom.make atom in
  let node1 = Term.Ref.make () in
  let node2 = Term.Ref.make () in
  Term.put_atom node1 atom1;
  Term.put_atom node2 atom2;
  Printf.printf "Compare: %i\n%!" (Term.compare node1 node2);
  Printf.printf "Equal: %b\n%!" (Term.equal node1 node2);
  Printf.printf "Unifiable: %b\n%!" (Term.unify node1 node2)
